web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,401 @@
+import * as React from "react";
+
+import SpaRenderer from "core/spa/spa-renderer";
+
+import { AsyncButton } from "components/buttons";
+import { Messages } from "components/messages";
+import { TopPanel } from "components/panels/TopPanel";
+
+enum SSLMode {
+  UseSSL = "use-ssl",
+  CreateSSL = "create-ssl",
+}
+
+type Props = {};
+
+type State = {
+  email: "";
+  sslMode: SSLMode;
+  loading: boolean;
+  maxSquidCacheSize: string;
+  messages: any[];
+  proxyFQDN: string;
+  success?: any;
+};
+
+class ProxyConfig extends React.Component<Props, State> {
+  initState: State;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.initState = {
+      email: "",
+      sslMode: SSLMode.CreateSSL,
+      loading: false,
+      maxSquidCacheSize: "",
+      messages: [],
+      proxyFQDN: "",
+    };
+
+    this.state = this.initState;
+  }
+
+  onSubmit = () => {
+    // this.setState({ messages: [], loading: true });
+    alert("TODO: let's do something on submit now");
+  };
+
+  clearFields = () => {
+    this.setState(this.initState);
+  };
+
+  proxyFQDNChanged = (event) => {
+    this.setState({
+      proxyFQDN: event.target.value,
+    });
+  };
+
+  maxSquidCacheSizeChanged = (event) => {
+    this.setState({
+      maxSquidCacheSize: event.target.value,
+    });
+  };
+
+  emailChanged = (event) => {
+    this.setState({
+      email: event.target.value,
+    });
+  };
+
+  sslModeChanged = (event) => {
+    this.setState({
+      sslMode: event.target.value,
+    });
+    console.log(event.target.value);
+  };
+
+  render() {
+    var messages: React.ReactNode = null;
```

Comment:
I would recommend splitting this, and possibly also `buttons` and `useSSLFields` into separate methods, maybe even files. Right now this render function is very large.